pkg/kubelet/kubelet.go

```diff
@@ -2393,8 +2432,12 @@ func (kl *Kubelet) generatePodStatus(pod *api.Pod) (api.PodStatus, error) {
 			}
 		}
 	}
-
-	podStatus.Conditions = append(podStatus.Conditions, getPodReadyCondition(spec, podStatus.ContainerStatuses)...)
+	existingStatus, ok := kl.statusManager.GetPodStatus(pod.UID)
```

Comment:
You may want to rebase because this function is slightly behind. GetPodStatus is already called in this function at HEAD.